src/defaults/default.js

```diff
@@ -36,36 +36,35 @@ function pipe(next, params, constants, logger) {
 const pre = cont => cont;
 
 /**
- * A standard cleanup function that takes OpenWhisk-style parameters and turns
- * them into an Express-style request object which is returned.
- * @param {Object} params Parameters following OpenWhisk convention, including
- * __ow_method and __ow_headers for HTTP requests
- * @returns {Object} A req object that is equivalent to an Express request object,
+ * A function that takes OpenWhisk-style req parameter and turns
+ * it into the original Express-style request object which is returned.
+ * @param {Object} payload Pipeline payload
+ * @param {Object} action Object representing the OpenWhisk action, including
+ * the request, __ow_method and __ow_headers for HTTP requests and secrets
+ * @returns {Object} The original req object that is equivalent to an Express request object,
  * including a headers, method, and params field
  */
-function adaptOWRequest(params) {
-  // use destructuring to drop __ow_headers and __ow_method from params
-  /* eslint-disable camelcase, no-underscore-dangle */
-  const { __ow_headers, __ow_method, ...newparams } = params;
-
-  return {
-    request: {
-      headers: __ow_headers,
-      params: newparams,
-      method: __ow_method,
-    },
-  };
-  /* eslint-enable: camelcase, no-underscore-dangle */
+function adaptOWRequest(payload, { logger, request: { params: { req = '{}' } } }) {
+  try {
+    return {
+      request: JSON.parse(req),
```

Comment:
This should be `querystring.parse(req)` to work with adobe/helix-cli#153 – and it would only work for the request parameters

Reply:
no. we need the entire request (method, url, params, headers)

Reply:
@trieloff I will commit this version to make progress. I guess we need to iterate again over how and what to pass into the external request object.